### PR TITLE
[Feature] Auto cordial

### PR DESCRIPTION
**Pull request by Jaksuhn**
It works just fine outside of the job checks on lines 71-72. It'll time out if LocalPlayer is null, and even when it isn't, with those lines present it will just be incredibly slow to ever work.

### CODE REVIEW

File: PandorasBox/Features/Actions/AutoCordial.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        };
+
+        private bool WillOvercap(int gp_recovery)
+        {
+            return Svc.ClientState.LocalPlayer.CurrentGp + gp_recovery < Svc.ClientState.LocalPlayer.MaxGp;
+        }
+
+        private void RunFeature(Framework framework)
+        {
+            if (!(Svc.ClientState.LocalPlayer.ClassJob.Id == 16) || !(Svc.ClientState.LocalPlayer.ClassJob.Id == 17)) return;
```
Comment (Taurenkey) [inline PandorasBox/Features/Actions/AutoCordial.cs:71]: Add null check for LocalPlayer before this

File: PandorasBox/Features/Actions/AutoCordial.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            uint watered_cordial = 16911;
+            uint watered_cordial_hq = 1016911;
+            List<uint> cordials = new List<uint>() { hi_cordial, reg_cordial_hq, reg_cordial, watered_cordial_hq, watered_cordial };
+
+            int hi_recovery = 400;
+            int reg_recovery_hq = 350;
+            int reg_recovery = 300;
+            int water_recovery_hq = 200;
+            int water_recovery = 150;
+            List<int> recoveries = new List<int>() { hi_recovery, reg_recovery_hq, reg_recovery, water_recovery_hq, water_recovery };
```
Comment (Taurenkey) [inline PandorasBox/Features/Actions/AutoCordial.cs:90]: Add this outside the run feature as this data is static

File: PandorasBox/Features/Actions/AutoCordial.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            }
+            ImGui.Checkbox("Invert Priority (Watered -> Regular -> Hi)", ref Config.InvertPriority);
+            ImGui.Checkbox("Prevent Overcap", ref Config.PreventOvercap);
+            ImGui.Checkbox("Use on Fisher", ref Config.UseOnFisher);
+        };
+
+        private bool WillOvercap(int gp_recovery)
+        {
+            return Svc.ClientState.LocalPlayer.CurrentGp + gp_recovery < Svc.ClientState.LocalPlayer.MaxGp;
+        }
```
Comment (Taurenkey) [inline PandorasBox/Features/Actions/AutoCordial.cs:67]: Consider looking to pass the item ID and check the sheets for a recovery amount.

File: PandorasBox/Features/Actions/AutoCordial.cs (hunk 1 above)
Comment (Jaksuhn) [reply PandorasBox/Features/Actions/AutoCordial.cs:90]: done in 1e66a08

File: PandorasBox/Features/Actions/AutoCordial.cs (hunk 3 above)
Comment (Jaksuhn) [reply PandorasBox/Features/Actions/AutoCordial.cs]: So I added a (commented out) draft of this in f74e13c just so you can see why I think this wouldn't work. The issue with this is that I don't believe recovery is present in the sheets, at least not as a regular value, just in the description. Also, would have to have more complicated handling in there for if the item is HQ, so I think the static values are probably the safer bet.

Comment (Jaksuhn) [reply PandorasBox/Features/Actions/AutoCordial.cs]: Something like this? Not really sure how the Data{HQ} column should be handled.
`internal static (string Name, uint Id, uint ItemActionID, uint NQGP, uint HQGP)[] cordials;`
```cordials = Svc.Data.GetExcelSheet<Item>()
      .Where(row => row.Name.ToString().Contains("cordial")
      .Select(row => (
          Name: row.Name,
          Id: row.RowId,
          ItemActionID: row.ItemAction,
          NQGP: Svc.Data.GetExcelSheet<ItemAction>()
              .First(actionRow => actionRow.RowId == row.ItemActionID).Data[0],
          HQGP: Svc.Data.GetExcelSheet<ItemAction>()
              .First(actionRow => actionRow.RowId == row.ItemActionID).Data{HQ}[0]
      )).ToArray());```

**Pull request merged**